linked_list/linked_list.py

```diff
@@ -4,72 +4,131 @@ class Node:
 
     def __init__(self, value, next_node = None):
         self.value = value
-        self.next = next_node
+        self.next_node = next_node
+
+    def set_next_node(self, next_node):
+        self.next_node = next_node
 
 # Defines the singly linked list
 class LinkedList:
     def __init__(self):
-      self.head = None # keep the head private. Not accessible outside this class
+        self.head = None # keep the head private. Not accessible outside this class
 
     # returns the value in the first node
     # returns None if the list is empty
     # Time Complexity: ?
     # Space Complexity: ?
     def get_first(self):
-        pass
+        if self.head:
+            return self.head.value
+        return None
 
 
     # method to add a new node with the specific data value in the linked list
     # insert the new node at the beginning of the linked list
     # Time Complexity: ?
     # Space Complexity: ?
     def add_first(self, value):
-        pass
+        new_node = Node(value)
+        new_node.set_next_node(self.head)
+        self.head = new_node
 
     # method to find if the linked list contains a node with specified value
     # returns true if found, false otherwise
     # Time Complexity: ?
     # Space Complexity: ?
     def search(self, value):
-        pass
+        current_node = self.head
+        while current_node:
+            if current_node.value == value:
+                return True
+            current_node = current_node.next_node
+        return False
 
     # method that returns the length of the singly linked list
     # Time Complexity: ?
     # Space Complexity: ?
     def length(self):

```

Comment:
👍 Time/space complexity?

I do want to point out that an alternative solution is to maintain a `length` field that you update when nodes are updated or deleted and maintain an O(1) time complexity for this method.